install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -727,13 +727,14 @@ data:
 {{- if $socketLB }}
 {{- if hasKey $socketLB "enabled" }}
   bpf-lb-sock: {{ $socketLB.enabled | quote }}
-  bpf-lb-sock-terminate-pod-connections: {{ $socketLB.enabled | quote }}
 {{- end }}
 {{- if hasKey $socketLB "hostNamespaceOnly" }}
   bpf-lb-sock-hostns-only: {{ $socketLB.hostNamespaceOnly | quote }}
 {{- end }}
 {{- if hasKey $socketLB "terminatePodConnections" }}
   bpf-lb-sock-terminate-pod-connections: {{ $socketLB.terminatePodConnections | quote }}
+{{- else }}
+  bpf-lb-sock-terminate-pod-connections: {{ $socketLB.enabled | quote }}
```

Comment:
The values file indicates that this should default to `enabled`, but the former logic defaults it to `$socketLB.enabled`. Not sure which is desired.

Reply:
It is also auto-enabled if hostServices are enabled, see https://github.com/cilium/cilium/blob/19de3f774778221303c16c7b1fc41befaaadc797/install/kubernetes/cilium/templates/cilium-configmap.yaml#L74